### PR TITLE
Added onEditingrowCancel prop

### PR DESCRIPTION
**Pull request by pjtilsworth**
I've updated the docs & crud example as well.  Please you as you see fit!

### CODE REVIEW

File: src/MaterialReactTable.tsx [hunk 1]
```diff
@@ -945,6 +945,7 @@ export type MaterialReactTableProps<TData extends Record<string, any> = {}> =
       table: MRT_TableInstance<TData>;
       values: Record<LiteralUnion<string & DeepKeys<TData>>, any>;
     }) => Promise<void> | void;
+    onEditingRowCancel?: OnChangeFn<MRT_Row<TData> | null>;
```
Comment (KevinVandy) [inline src/MaterialReactTable.tsx:948]: This won't be the `onChange<>` type, will need to change it probably

File: material-react-table-docs/pages/docs/guides/editing.mdx [hunk 2]
```diff
@@ -70,6 +71,8 @@ You must provide an `onEditingRowSave` callback function prop where you will get
 
 > The `onEditingRowSave` callback function prop includes a `exitEditingMode` parameter that must be called in order to exit editing mode upon save. The reason for this is so that you can perform validation checks before letting the modal close.
 
+If you are implementing validation, you may need to use `onEditingRowCancel` to clear the validation error state.  This is demonstrated in the [Editing (Crud)](/docs/examples/editing-crud) example.
```
Comment (KevinVandy) [inline material-react-table-docs/pages/docs/guides/editing.mdx:74]: I'll probably move this documentation down to the validation section below

File: material-react-table-docs/components/prop-tables/rootProps.ts [hunk 3]
```diff
@@ -1474,6 +1474,16 @@ export const rootProps: PropRow[] = [
...
   },
+  {
+    propName: 'onEditingRowCancel',
+    defaultValue: '',
+    description: '',
+    link: '',
+    linkText: '',
+    required: false,
+    source: 'MRT',
+    type: 'OnChangeFn<MRT_Row<TData> | null>',
```
Comment (KevinVandy) [inline material-react-table-docs/components/prop-tables/rootProps.ts:1485]: type will change to else more accurate

**Review by KevinVandy — APPROVED** [review]

File: src/MaterialReactTable.tsx (hunk 1 above)
Comment (pjtilsworth) [reply src/MaterialReactTable.tsx:948]: Yeah ....`onChange<>` maybe not best type for describing what is happening.  I thought returning the row might be some value possibly for messaging/logging purposes.

**Pull request merged**